presto-docs/src/main/sphinx/optimizer/push-down.rst

```diff
@@ -0,0 +1,66 @@
+=========
+Push Down
+=========
+
+Presto can push down processing queries, or segments of queries, down to the
+connected data source. This means that a specific function or other operation is
+passed through to the underlying database for processing. An improved overall query performance can be the result of this push down.
+
+Support for push down is specific to each connector and the relevant underlying
+database or storage system.
+
+Every Presto aggregate function is eligible for push down, as long as it
+operates on types supported by the given connector.
+
+Analysis and Confirmation
+-------------------------
+
+Push down depends on a number of factors:
+
+* generic support for push down in the connector
+* function or operation specific support for push down in the connector
+* query that allows the detection of the function to push down
+
+The best way to analyze if push down for a specific query is performed, is to
+take a closer look at the :doc:`EXPLAIN plan </sql/explain>` of the query. If an
+operations such as an aggregate function successfully pushed down to the
+connector, the explain plan does **not** show that operator. The explain plan
```

Comment:
Is that true? If the pushdown result is not guaranteed by the connector, Presto should keep the filter. For example, pushdown of limit is not guaranteed if `LimitApplicationResult.limitGuaranteed` is false. In such a case, the explain plan would show the operator while the pushdown happens in the connector.

Reply:
Not sure .. we need some user facing docs for this and this is my attempt to get this started.

Reply:
@findepi can you confirm what I wrote or provide corrected info?

Reply:
@Lewuathe this is true for _filters_ and JDBC connectors (although I am improving this in https://github.com/prestosql/presto/pull/4565)
however, there is no way to "improve aggregation", connector must do this thing correct, or not do this at all

@mosabua yes, pushded down agg won't show up in EXPLAIN anymore